### ISSUE TITLE
Cannot run OpenAPI code-generation for CRD with `operator-sdk v0.12.0`

### ISSUE THREAD

**Issue by longkb** [metal3-io/baremetal-operator#360]
I am trying to run `make generate` in PR #292 for CRD code-generation with `operator-sdk v0.12.0`. But I got an error

```bash
$ make generate
operator-sdk generate k8s
INFO[0000] Running deepcopy code-generation for Custom Resource group versions: [metal3:[v1alpha1], ]
INFO[0001] Code-generation complete.
operator-sdk generate openapi
INFO[0000] Running OpenAPI code-generation for Custom Resource group versions: [metal3:[v1alpha1], ]
Error: error generating CRD for Group metal3 Version v1alpha1 Kind BareMetalHost
Usage:
  operator-sdk generate openapi [flags]

Flags:
  -h, --help   help for openapi

Global Flags:
      --verbose   Enable verbose logging

Makefile:41: recipe for target 'generate' failed
make: *** [generate] Error 1
```

**Comment by jan-est** [comment]
What is your operator-sdk version? Version we need to run make generate successfully is:
operator-sdk version: "v0.12.0"

**Comment by longkb** [comment]
> 
> 
> What is your operator-sdk version? Version we need to run make generate successfully is:
> operator-sdk version: "v0.12.0"

I tried with `v0.12.0`, but it was not work

**Comment by longkb** [comment]
> 
> 
> > What is your operator-sdk version? Version we need to run make generate successfully is:
> > operator-sdk version: "v0.12.0"
> 
> I tried with `v0.12.0`, but it was not work

Sorry. I have tried again.
Your patch #349 can work with `master` branch
But when I run with my PR (#292), it reach an error with `openapi` for Custom Resource

```
$ make test
operator-sdk generate k8s
INFO[0000] Running deepcopy code-generation for Custom Resource group versions: [metal3:[v1alpha1], ]
INFO[0001] Code-generation complete.
operator-sdk generate openapi
INFO[0000] Running OpenAPI code-generation for Custom Resource group versions: [metal3:[v1alpha1], ]
Error: error generating CRD for Group metal3 Version v1alpha1 Kind BareMetalHost
Usage:
  operator-sdk generate openapi [flags]

Flags:
  -h, --help   help for openapi

Global Flags:
      --verbose   Enable verbose logging

Makefile:41: recipe for target 'generate' failed
make: *** [generate] Error 1

```

**Comment by jan-est** [comment]
@longkb yes my patch is aim to work with master. Can you run PR https://github.com/metal3-io/baremetal-operator/pull/292 by installing code-generator like you proposed in you PR https://github.com/metal3-io/baremetal-operator/pull/361?

**Comment by longkb** [comment]
> 
> 
> @longkb yes my patch is aim to work with master. Can you run PR #292 by installing code-generator like you proposed in you PR #361?

It still got an error. The results show that
- My PR #361 only worked with operator-sdk v0.8.0
- operator-sdk v0.12.0 stills got the error as show above [1]

Is there any problem with ` Version v1alpha1` of baremetalhost object?

[1] logs
```bash
$ make generate
operator-sdk generate k8s
INFO[0000] Running deepcopy code-generation for Custom Resource group versions: [metal3:[v1alpha1], ]
INFO[0001] Code-generation complete.
operator-sdk generate openapi
INFO[0000] Running OpenAPI code-generation for Custom Resource group versions: [metal3:[v1alpha1], ]
Error: error generating CRD for Group metal3 Version v1alpha1 Kind BareMetalHost
Usage:
  operator-sdk generate openapi [flags]

Flags:
  -h, --help   help for openapi

Global Flags:
      --verbose   Enable verbose logging

Makefile:41: recipe for target 'generate' failed
make: *** [generate] Error 1

```

**Comment by jan-est** [comment]
What files you have in your Baremetal Operator deploy/crds/ folder? If there is any files other than metal3.io_baremetalhosts_crd.yaml my "make generate" will end up with error.

**Comment by stbenjam** [comment]
Did this ever get resolved for you?

**Comment by kirankt** [comment]
OK. I was able to replicate this issue. The problem seems to be here: https://github.com/metal3-io/baremetal-operator/blob/3d7a06eb80b93e0b2df5a4d5aa9cc211eabf61df/pkg/apis/metal3/v1alpha1/baremetalhost_types.go#L393 and https://github.com/metal3-io/baremetal-operator/blob/3d7a06eb80b93e0b2df5a4d5aa9cc211eabf61df/pkg/apis/metal3/v1alpha1/baremetalhost_types.go#L398 the kubebuilder validation marker for Enum should be a ';' rather than ','.

**Comment by zaneb** [comment]
Has that changed recently? The examples in the docs definitely have `,`: https://book-v1.book.kubebuilder.io/beyond_basics/generating_crd.html

**Comment by kirankt** [comment]
> Has that changed recently? The examples in the docs definitely have `,`: https://book-v1.book.kubebuilder.io/beyond_basics/generating_crd.html

Huh. Strange. I see the use of ';' in this book: https://book.kubebuilder.io/reference/generating-crd.html. operator-sdk v0.12.0 silently quits without any error, when using a ',' instead of a ';' for enums.

**Comment by kirankt** [comment]
> > Has that changed recently? The examples in the docs definitely have `,`: https://book-v1.book.kubebuilder.io/beyond_basics/generating_crd.html
> 
> Huh. Strange. I see the use of ';' in this book: https://book.kubebuilder.io/reference/generating-crd.html. operator-sdk v0.12.0 silently quits without any error, when using a ',' instead of a ';' for enums.

Because v0.12.0 silently quits without spitting any actual problem, I had to use v0.14.0 to make sense of the error. Of course, v0.14.0 opens an entirely different can of worms. e.g it'll complain about use of float64 in the API.

**Comment by zaneb** [comment]
Oh, book-v1 is the "legacy documentation", but still sometimes shows up first in searches. Great.

**Comment by kirankt** [comment]
Should be fixed in #395. We can close this issue.
